fcrepo-client-impl/src/main/java/org/fcrepo/client/impl/FedoraRepositoryImpl.java

```diff
@@ -175,7 +185,7 @@ public FedoraDatastream createOrUpdateRedirectDatastream(final String path, fina
             final int statusCode = status.getStatusCode();
 
             if (statusCode == SC_CREATED) {
-                return getDatastream(path);
+                return getDatastream(prependTransactionId(path));
```

Comment:
Although this scenario is recognized and ignored in the implementation of `prependTransactionId(..)`, is there ever a case where this apparently redundant call to `prependTransactionId(..)` is not redundant? If not, I would suggest removing the update to this line.